test/powershell/engine/Api/TypeInference.Tests.ps1

```diff
@@ -650,8 +650,10 @@ Describe "Type inference Tests" -tags "CI" {
     }
 
     It "Infers type from variable with AllowSafeEval" {
-        function Hide-GetProcess { Get-Process }
-        $p = Hide-GetProcess
+        # Invoke-Expression is used to "hide" Get-Process from the type inference.
+        # If the typeinference code is updated to handle Invoke-Expression, this test will need to find some other way to set $p
+        # so that the type inference can't figure it out without evaluating the variable value
+        $p = Invoke-Expression -Command 'Get-Process'
```

Comment:
Can we use the pattern
https://github.com/PowerShell/PowerShell/blob/8733c1793eaf1cd99208c921ab58786a7039bee0/test/powershell/engine/Api/TypeInference.Tests.ps1#L1569-L1575

Reply:
No because the type inference based on the variable runtime value is a last resort: https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/parser/TypeInferenceVisitor.cs#L2485  
The regular type inference will still see the variable being assigned by the command `Get-Process` and hence, use the output type defined there.
I think the only way to make the test fully robust is to make a test specific `Get-Process2` cmdlet without output types defined so that the type inference has no way of finding out the value without the SafeEval.